### DISCUSSION TITLE
Documentation on trigger rule for `one_success` misleading

### DISCUSSION THREAD

**Discussion by MondeSinxi** [apache/airflow#36055] [Q&A]
### What do you see as an issue?

https://airflow.apache.org/docs/apache-airflow/2.7.3/core-concepts/dags.html#trigger-rules

The trigger rule for `one_success` states that " At least one upstream task has succeeded (does not wait for all upstream tasks to be done)". I believe that this should be changed to "At least one parent task has succeeded...". In the code snippet below I have a `task_f`, the trigger rule is set to `one_success`, I expect that once task `a` has succeeded (as on of the upstream tasks of `f`) then `f` should also run. However, the task is marked as `upstream_failed` as soon as task `c` fails. The expected behavior does not line up with the documentations.

```
import time
from datetime import datetime
from airflow import DAG
from airflow.operators.python import PythonOperator
from airflow.utils.trigger_rule import TriggerRule


def wait():
    time.sleep(3)


def fail():
    raise Exception("I should fail!")


with DAG("example", schedule=None, start_date=datetime(2023, 11, 30)) as dag:
    task_a = PythonOperator(task_id="a", python_callable=wait)
    task_b = PythonOperator(task_id="b", python_callable=wait)
    task_c = PythonOperator(task_id="c", python_callable=fail)
    task_d = PythonOperator(task_id="d", python_callable=wait)
    task_e = PythonOperator(task_id="e", python_callable=wait)
    task_f = PythonOperator(task_id="f", python_callable=wait, trigger_rule=TriggerRule.ONE_SUCCESS)

    task_a >> task_b >>  task_c >> task_d >> task_f

    task_c >> task_e >> task_f
```

![image](https://github.com/apache/airflow/assets/30410358/bae71952-c0fe-42db-a889-a6036b3fbc2c)

### Solving the problem

I believe that this should be changed to "At least one parent task has succeeded...".

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template! If you are willing to raise PR to address this issue please do so, no need to wait for approval.

**Comment by eladkal** [comment]
trigger rules check only **direct** upstream. Task `f` trigger rule will not consider tasks `c, b`
